simphony_mayavi/plugins/engine_manager.py

```diff
@@ -0,0 +1,95 @@
+from traits.api import HasTraits, Instance, ListStr, Str, Dict, Property
+from traitsui.api import View, Group, Item
+
+from mayavi.core.trait_defs import DEnum
+
+from simphony.cuds.abc_modeling_engine import ABCModelingEngine
+
+
+class EngineManager(HasTraits):
+    ''' A basic container of Simphony Engine that comes with a GUI.
+
+    Additional panel can be added to support more operations related
+    to the modeling engines
+
+    Attributes
+    ----------
+    engines : dict
+        Mappings of Simphony Modeling Engines in this manager
+    engine_name : str
+        Name of the Simphony Modeling Engine
+    engine : Instance of ABCModelingEngine
+        Simphony Modeling Engine
+    '''
+    # Engines that are added to the manager
+    engines = Dict(Str, Instance(ABCModelingEngine))
+
+    # Names of engines in the Manager
+    _engine_names = ListStr
+
+    # Selected engine (overloaded BasicPanel)
+    engine = Property
```

Comment:
for the property, you tycally pass the type of the property and what it depends on